presto-main/src/main/java/com/facebook/presto/connector/system/jdbc/FilterUtil.java

```diff
@@ -22,15 +22,15 @@
 
 import java.util.Optional;
 
+import static com.google.common.base.Preconditions.checkArgument;
+
 final class FilterUtil
 {
     private FilterUtil() {}
 
     public static Optional<String> stringFilter(TupleDomain<Integer> constraint, int index)
     {
-        if (constraint.isNone()) {
```

Comment:
Doesn't code rely on this semantics? Can't constraint be none? Shouldn't callers be fixed too?